fix: tsc --noEmit fails in clean checkout — @spawnforge/ui unresolvable without dist/

## Problem

Running `tsc --noEmit` in the `web/` directory on a clean checkout (where `packages/ui` has never been built) emits three type errors:

```
src/components/editor/EditorLayout.tsx(20,16): error TS2307: Cannot find module '@spawnforge/ui'
src/components/settings/AppearanceTab.tsx(3,55): error TS2307: Cannot find module '@spawnforge/ui'
src/components/settings/AppearanceTab.tsx(20,29): error TS7006: Parameter 't' implicitly has an 'any' type.
```

The third error is a downstream consequence of the first two — once `THEME_NAMES` is unresolvable, TypeScript can't infer the map callback type.

**Root cause:** `packages/ui/package.json` exports `types: ./dist/index.d.ts` (requires a build), but also has a `development: ./src/index.ts` condition that points directly to source. The `web/tsconfig.json` did not set `customConditions: ["development"]`, so `tsc` used the `types` condition and failed when `dist/` was absent.

`vitest.config.jsdom.ts` already sets `conditions: ['development', ...]` so tests pass in all environments — it was only `tsc` that was broken.

## Also

`@vitest/coverage-v8` was only installed in `web/node_modules`, not hoisted to root. Running coverage via the root `vitest` binary caused `ERR_MODULE_NOT_FOUND` for `@vitest/coverage-v8`.

## Fix

See PR — adds `"customConditions": ["development"]` to `web/tsconfig.json` and hoists `@vitest/coverage-v8` to root `package-lock.json`.

--- a/web/package.json
+++ b/web/package.json
@@ -101,7 +101,7 @@
     "@types/node": "^25",
     "@types/react": "^19",
     "@types/react-dom": "^19",
-    "@vitest/coverage-v8": "^4.1.4",
+    "@vitest/coverage-v8": "^4.1.5",
     "ajv": "^6.14.0",
     "drizzle-kit": "^0.31.10",
     "eslint": "^9",

--- a/web/tsconfig.json
+++ b/web/tsconfig.json
@@ -28,6 +28,10 @@
     // where array access is guaranteed safe (e.g., after length checks, Map.get with existence guard).
     "noUncheckedIndexedAccess": false,
     "incremental": true,
+    // Resolve @spawnforge/ui from TS source (via "development" export condition)
+    // so tsc works in clean checkouts where packages/ui hasn't been built yet.
+    // Matches the conditions list in vitest.config.jsdom.ts.
+    "customConditions": ["development"],
     "plugins": [
       {
         "name": "next"

